Doc/library/sqlite3.rst

```diff
@@ -2264,13 +2343,84 @@ can be found in the `SQLite URI documentation`_.
 Explanation
 -----------
 
+.. _sqlite3-transaction-control:
 .. _sqlite3-controlling-transactions:
 
 Transaction control
 ^^^^^^^^^^^^^^^^^^^
 
-The :mod:`!sqlite3` module does not adhere to the transaction handling recommended
-by :pep:`249`.
+:mod:`!sqlite3` offers multiple methods of controlling whether,
+when and how database transactions are opened and closed.
+:ref:`sqlite3-transaction-control-autocommit` is recommended,
+while :ref:`sqlite3-transaction-control-isolation-level`
+is consistent with pre-Python 3.12 versions of the :mod:`!sqlite3` module.
+
+.. _sqlite3-transaction-control-autocommit:
+
+Transaction control via the ``autocommit`` attribute
+""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+The recommended way of controlling transaction behaviour is through
+the :attr:`Connection.autocommit` attribute,
+which should preferrably be set using the *autocommit* parameter
+of :func:`connect`.
+
+It is suggested to set *autocommit* to ``False``,
+which implies :pep:`249`-compliant transaction control.
+This means:
+
+* :mod:`!sqlite3` ensures that a transaction is always open,
+  so :meth:`Connection.commit` and :meth:`Connection.rollback`
+  will implicitly open a new transaction immediately after closing
+  the pending one.
+  :mod:`!sqlite3` uses ``BEGIN DEFERRED`` statements when opening transactions.
+* Transactions should be committed explicitly using :meth:`!commit`.
+* Transactions should be rolled back explicitly using :meth:`!rollback`.
+* An implicit rollback is performed if the database is
+  :meth:`~Connection.close`-ed with pending changes.
+
+Set *autocommit* to ``True`` to enable SQLite's `autocommit mode`_.
+In this mode, :meth:`Connection.commit` and :meth:`Connection.rollback`
+have no effect.
+
+Set *autocommit* to :data:`LEGACY_TRANSACTION_CONTROL`
+to leave transaction control behaviour to the
+:attr:`Connection.isolation_level` attribute.
+See :ref:`sqlite3-transaction-control-isolation-level` for more information.
+
+.. note::
+
+   :attr:`~Connection.autocommit` and SQLite's `autocommit mode`_
+   are not linked.
+   For example, explicitly issuing a ``BEGIN`` statement when
+   :attr:`!autocommit` is ``True`` will not change the value of
+   the *autocommit* attribute.
+
+   Use :attr:`Connection.in_transaction` to query the low-level SQLite
+   autocommit mode.
+
+.. note::
+
+   The :attr:`Connection.isolation_level` attribute has no effect unless
+   :attr:`Connection.autocommit` is :data:`LEGACY_TRANSACTION_CONTROL`.
```

Comment:
IMO, its more salient to call this out in the `isolation_level` section in a note, as opposed to one here, especially since it makes the document structure hard to parse with a note, then a note, then a heading, then another note. I'd suggest embedding it in the existing note at the top of the next `isolation_level` section.